### DISCUSSION TITLE
merge json values for integer templated value

### DISCUSSION THREAD

**Discussion by aeb-dev** [hairyhenderson/gomplate#1738] [Q&A]
Sorry for the title. I did not know what to write because I am combining multiple features and tools together.

First of all thanks for the tool. I will remember to use it in my future endeavors.

Here is my case; I am trying to combine json files to using gomplate. After the files are combine it will be directed to vault agent so that it can also fetch and render secrets to it. I can not use `gomplate`'s vault feature because, I am using kubernetes vault injector so that the secrets will be reloaded automatically. Here is the problem, 

I have common configs for service like the following, this file is injected from `context` with name `configurator`:
```json
{
     "Timeout": 5000,
}
```

I have a big base `appsettings.json`:

```json
{
  "Configuration": {
    "Timeout": {{{ .configurator.Timeout }}}
  },
  "Connection": {
    "Port": {{ integer port value inside vault that will be rendered by vault agent }}
  }
}
```

I have a template with patching support like following:
```
{{{ $base := file.Read "path/to/base/appsettings.json" | tmpl.Inline | json }}}
{{{- define "patch" -}}}
{
  "Configuration": {
    "Timeout": 5001
  }
}
{{{- end -}}}

{{{ merge (tmpl.Exec "patch" | json) $base | toJSONPretty "  " }}}
```

If I render this since "Connection.Port" is not a valid json value gomplate will result in error. If I surround it with "" it works but then it will not be an integer anymore and it breaks some services.

So my question is how can get I around this issue? Is there anyway to tell gomplate to ignore certain values?

Also, If you think this flow can be improved in anyway I am open to suggestions. As you probably understood already, I am trying to make a configuration management for our services.

**Comment by hairyhenderson** [comment]
> So my question is how can get I around this issue? Is there anyway to tell gomplate to ignore certain values?

not really - the issue is it's being parsed as JSON, when it isn't actually valid JSON.

Perhaps the solution could involve patching in the non-JSON value at the end? I don't have time to think deeply about how that could work, but hopefully it can get you started...
